### ISSUE TITLE
🐛 Bug: v1.1.x raised engines.node from 16 to 18

### ISSUE THREAD

**Issue by shiftj18** [JoshuaKGoldberg/ts-api-utils#389]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

The change in the 'engines.node' configuration from 16 to 18 uses the 'x' major version, e.g. 2.0.0.

And, rollback 1.1.0 and 1.2.0.

### Actual

![image](https://github.com/JoshuaKGoldberg/ts-api-utils/assets/8348072/096e3a50-f704-4ec9-9fd1-47a6d738ed4f)

![image](https://github.com/JoshuaKGoldberg/ts-api-utils/assets/8348072/b9e2bb34-5fec-42cf-8351-ddf018b8a4bc)

Shouldn't the node version be compatible with the x of the semver rule?

This causes packages or apps using the '^1.0.1' version to abort unexpectedly, when installed using the yarn(v1) package manager in an earlier version of the node environment. Eventually, the installation failed, and the directory 'node_modules' is not exists.

```bash
"error ts-api-utils@1.2.0: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.20.0" error Found incompatible module.'
```

**Comment by shiftj18** [comment]
![image](https://github.com/JoshuaKGoldberg/ts-api-utils/assets/8348072/aa383c4d-61ca-4181-b1a5-ec794d5edc75)

**Comment by aovchinn** [comment]
encountered this through typescript-eslint packages, on node 16

**Comment by kyle-revio** [comment]
Also ran into this now

**Comment by JoshuaKGoldberg** [comment]
Sorry folks, this is a bug. I'll fix it.

**Comment by JoshuaKGoldberg** [comment]
Should be fixed in v1.2.1. Sorry for the disruption.

```shell
npm i ts-api-utils@latest
```

**Comment by kyle-revio** [comment]
Thanks a million @JoshuaKGoldberg !
